### PR TITLE
added common dialog form and createAccount form

### PR DESCRIPTION
**Pull request by seanheinrichs**
Current Changes:

- Make Login page more friendly for tablet (login components share half the screen with image).
- Create `FormDialog.js` component. A common form dialog to be reused across the whole application.
- `CreateAccountFormDialog.js` component. A form so users can create accounts on the `LoginPage`.
- Create the `postUserEvent` epic to create users in the database.
- Implement `joi` validation on the front. It is connected through the common `FormDialog.js` component for easy reuse.

Desktop/Tablet
image

Mobile
image

### CODE REVIEW

File: frontend/src/common/components/FormDialog.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+	return (
+		<Dialog fullWidth maxWidth={maxWidth} open={open} onClose={close}>
+			<DialogTitle>{title}</DialogTitle>
+			<DialogContent>
+				<Form
+					onSubmit={onSubmit}
+					validate={generateErrors}
+					render={({ handleSubmit }) => (
+						<form onSubmit={handleSubmit} noValidate>
```
Comment (seanheinrichs) [inline frontend/src/common/components/FormDialog.js:39]: If you're curious about the final form syntax you can check out the following links:

- https://www.youtube.com/watch?v=Z8W102KMA-E
- https://codesandbox.io/s/react-final-form-material-ui-example-tqv09?file=/src/index.jsx:550-807

File: frontend/src/common/components/FormDialog.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	open,
+	onCancel,
+	onSubmit,
+	primaryButtonLabel,
+	secondaryButtonLabel,
+	title,
+	validationSchema
+}) {
+
+	const generateErrors = (values) => ValidationService(values, validationSchema).errors;
```
Comment (seanheinrichs) [inline frontend/src/common/components/FormDialog.js:29]: Whenever you create a form and want to use this common component, you will also need to add a joi validation schema (see mine below) or following docs:

- https://github.com/hapijs/joi

File: frontend/src/forms/CreateAccountFormDialog.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+
+// Redux
+import { useDispatch } from 'react-redux';
+import { postUserRequest } from '../state/user/userActions';
+
+// Form
+import { TextField } from 'mui-rff';
```
Comment (seanheinrichs) [inline frontend/src/forms/CreateAccountFormDialog.js:8]: To use material ui and react-final-form together you need to wrap the components. This library just does it for you. You can export all of the Material UI wrappers from this library `mui-rff`:

`{ TextField, Checkbox, Radio, Select }` ... etc.

File: frontend/src/forms/CreateAccountFormDialog.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import FormDialog from '../common/components/FormDialog';
+import createAccountSchema from './CreateAccountFormDialog.schema';
+
+// Util
+import queryString from 'query-string'
```
Comment (seanheinrichs) [inline frontend/src/forms/CreateAccountFormDialog.js:13]: too lazy to make my method to convert from key values to query string param. found this little library and embraced the power of javascript

**Comment by mdesson** [comment]
Great writeup! I added yaroslav as a reviewer so he can maybe take a look. I'll review later today or tomorrow.

**Comment by mdesson** [comment]
Alright, so I pushed a fix to the environment, so `make run` will work properly. No need to rebuild to get it to work. It was all in the docker-compose.yml and Makefile config.

File: frontend/src/forms/CreateAccountFormDialog.schema.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import Joi from '@hapi/joi';
+
+const createAccountSchema = Joi.object({
+	firstName: Joi.string().required().label('First Name'),
+	lastName: Joi.string().required().label('Last Name'),
+	email: Joi.string()
+		.email({ minDomainSegments: 2, tlds: { allow: ['com', 'net'] } })
```
Comment (mdesson) [inline frontend/src/forms/CreateAccountFormDialog.schema.js:7]: Would it be possible to use regex to just accept all emails that end in two or more letters? 

Like this: `[a-zA-Z]{2,}`

Comment (seanheinrichs) [reply frontend/src/forms/CreateAccountFormDialog.schema.js]: Had to follow JOI docs, but changed it so we (should) accept all emails.

**Pull request merged**